components/announcement_bar/freemium_trial_ended_announcement_bar/index.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {t} from 'utils/i18n';
+
+import AnnouncementBar from '../default_announcement_bar';
+import {AnnouncementBarTypes, Preferences, CloudBanners} from 'utils/constants';
+import useGetUsageDeltas from 'components/common/hooks/useGetUsageDeltas';
+import useGetLimits from 'components/common/hooks/useGetLimits';
+import {GlobalState} from 'types/store';
+import {isSystemAdmin} from 'mattermost-redux/utils/user_utils';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import useGetSubscription from 'components/common/hooks/useGetSubscription';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {
+    getCurrentUser,
+} from 'mattermost-redux/selectors/entities/users';
+import {anyUsageDeltaValueIsNegative} from 'utils/utils';
+import './freemium_trial_ended_announcement_bar.scss';
+
+const CloudFreemiumTrialEndAnnouncementBar: React.FC = () => {
+    const usageDeltas = useGetUsageDeltas();
+    const limits = useGetLimits();
+    const subscription = useGetSubscription();
+    const getCategory = makeGetCategory();
+    const dispatch = useDispatch();
+    const preferences = useSelector((state: GlobalState) =>
+        getCategory(state, Preferences.FREEMIUM_TRIAL_END_BANNER),
+    );
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const shouldShowBanner = () => {
+        if (!subscription) {
+            return false;
+        }
+
+        if (!limits[1]) {
+            return false;
+        }
+
+        if (!preferences) {
+            return false;
+        }
+        if (preferences.some((pref) => pref.name === CloudBanners.HIDE && pref.value === 'true')) {
+            return false;
+        }
+
+        const isFreeTrial = subscription.is_free_trial === 'true';
+        if (isFreeTrial) {
+            return false;
+        }
+
+        const trialEnd = new Date(subscription.trial_end_at * 1000);
+        const now = new Date();
+
+        // trial_end_at values will be 0 for all freemium subscriptions after June 6
+        // Subscriptions created prior to that will almost always have a trial_end_at value, guaranteed.
+        if (subscription.trial_end_at === 0 || trialEnd > now) {
```

Comment:
It seems that this will also not show the banner if the trial ended today and the admin logs in tomorrow ?

Reply:
a 0 trial_end_at means they've never had a trial. trialEnd being in the future means they are likely still on a trial. Unless I'm misunderstanding you?

Reply:
Sorry, I was the one that was confused - the comment referring to June 6th threw me off. 

Where do we guard for the ones that their trial has already past (aka current Starter customers)?

Reply:
I see your point, don't love the date check but don't have any better suggestion either